### PR TITLE
Create a PR comment with docs preview link

### PR DESCRIPTION
**Pull request by logankilpatrick**
It looks like per @ToucheSir 's comment here: #1784, there is already a documentation preview workflow setup. This simply adds a comment to the PR with a link like we do on the JuliaLang.org website repo.

### CODE REVIEW

**Comment by ToucheSir** [comment]
The problem is, AIUI, that docs previews are only generated for PRs opened by members (with the right set of permissions). So we couldn't tack this onto any old PR.

**Comment by logankilpatrick** [comment]
Yeah, it would be nice if it let us do that, the current version in this PR only runs when the PR is made from a branch, not from forks. So it will work as is.

File: .github/workflows/pr_comment.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: pr_comment
+on:
+  pull_request:
+    types: [opened]
+jobs:
+  pr_comment:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create PR comment
+        if: github.event_name == 'pull_request' && github.repository == github.event.pull_request.head.repo.full_name # if this is a pull request build AND the pull request is NOT made from a fork
```
Comment (logankilpatrick) [inline .github/workflows/pr_comment.yml:10]: @ToucheSir

Comment (ToucheSir) [reply .github/workflows/pr_comment.yml:10]: That's what I get for not scrolling to the end, thanks!

**Review by ToucheSir — APPROVED** [review]

**Pull request merged**